hibernate-gorm/src/main/groovy/io/micronaut/configuration/hibernate/gorm/HibernateDatastoreFactory.groovy

```diff
@@ -86,4 +88,22 @@ class HibernateDatastoreFactory {
         return datastore
     }
 
+    @Singleton
+    SessionFactory sessionFactory(HibernateDatastore hibernateDatastore) {
+        hibernateDatastore.getSessionFactory()
+    }
+
+    @Singleton
+    DataSource dataSource(HibernateDatastore hibernateDatastore) {
+        ((HibernateConnectionSource) hibernateDatastore.getConnectionSources().defaultConnectionSource).getDataSource()
+    }
+
+    @Primary
+    @Named("hibernate")
+    @Singleton
+    PlatformTransactionManager transactionManager(HibernateDatastore hibernateDatastore) {
+        hibernateDatastore.getTransactionManager()
+    }
+
+
```

Comment:
Had to re-add these Singleton bean methods because I was experiencing transactional issues in actual usage otherwise. Guessing something is done at compile-time and simply registering/injecting the beans via the ApplicationContext is insufficient.